android/src/main/java/com/onesignal/rnonesignalandroid/RNUtils.java

```diff
@@ -284,6 +284,54 @@ public static HashMap<String, String> convertReadableMapIntoStringMap(ReadableMa
         return stringMap;
     }
 
+    public static Map<String, Object> convertReadableMapToMap(ReadableMap readableMap) {
+        Map<String, Object> map = new HashMap<>();
+        ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+    
+        while (iterator.hasNextKey()) {
+            String key = iterator.nextKey();
+            ReadableType type = readableMap.getType(key);
+            map.put(key, convertValue(type, readableMap, key));
+        }
+    
+        return map;
+    }
+    
+    public static List<Object> convertReadableArrayToList(ReadableArray readableArray) {
+        List<Object> list = new ArrayList<>();
+    
+        for (int i = 0; i < readableArray.size(); i++) {
+            ReadableType type = readableArray.getType(i);
+            list.add(convertValue(type, readableArray, i));
+        }
+    
+        return list;
+    }
+    
+    private static Object convertValue(ReadableType type, ReadableMap map, String key) {
+        switch (type) {
+            case Null: return null;
+            case Boolean: return map.getBoolean(key);
+            case Number: return map.getDouble(key);
+            case String: return map.getString(key);
+            case Map: return convertReadableMapToMap(map.getMap(key));
+            case Array: return convertReadableArrayToList(map.getArray(key));
+            default: throw new IllegalArgumentException("Unknown type for key: " + key);
```

Comment:
I don't think we should throw, are there other types that we aren't accounting for? We should be consistent when input is invalid. For example, in the initial method call from Javascript, we log error and no-op. `public void trackEvent(String name, @Nullable ReadableMap properties)` could have the same behavior?

Reply:
I don't think Java allows you to compile without the default case as you must return an Object in any code path (or throw)